### DISCUSSION TITLE
Appdata is not present

### DISCUSSION THREAD

**Discussion by Joly0** [nextcloud/all-in-one#664] [Questions] [answered]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. Configure volumes for master container
2. Start master container
3. Setup everything and let the master container start all the child containers

### Expected behavior 
All containers should startup
### Actual behavior 
Following log output from nextcloud container 
`psql: error: connection to server at "nextcloud-aio-database" (172.20.0.5), port 5432 failed: FATAL:  password authentication failed for user "oc_nextcloud"
Configuring Redis as session handler
Appdata is not present. Did you maybe change the datadir after aio was first started?`

### Host OS 
Unraid / Slackware

#### Nextcloud AIO version 
Nextcloud AIO v1.2.0

#### Current channel 
latest

#### Other valuable info 
I changed the docker-aio-config folder for the nextcloud-aio-mastercontainer so i could store these files in a safe directory, aswell added the environment variable for the apache port as shown in the documentation and i added an environment variable to the container for the data dir.

I guess the issue comes from me settings a custom folder for the docker-aio-config files, which all the other containers dont use, they try to use the default one (this should maybe get changed imo if its the issue)

EDIT:

I have deleted everything and started this from scratch and now i get the following in the nextcloud_aio_nextcloud container logs:

`Configuring Redis as session handler
Appdata is not present. Did you maybe change the datadir after aio was first started?`

**Comment by szaimen** [comment, marked as answer]
I think you did not completely reset the instance before changing values...
I'll add some docs how to reset it corrrctly...

**Reply by Joly0** [reply]
I edited my initial issue. I have deleted everything properly, removed all the volumes, images and all data that was created. I then added a data directory, gave it the correct permissions and startet the mastercontainer with the correct settings and now i get this error:
`Configuring Redis as session handler
Appdata is not present. Did you maybe change the datadir after aio was first started?`

**Reply by szaimen** [reply]
I see. Was the chosen target data directory empty  before you tried to reinstall it?

**Reply by Joly0** [reply]
yes, it was empty

**Reply by szaimen** [reply]
I fear you need to try again as I cannot reproduce this. However I will add some documentation on how to reset it properly.

**Reply by Joly0** [reply]
Got it working. i removed everything again, all the containers, images, volumes. Created the data directory with chown and chmod, configured and setup the mastercontainer and let it do its things and after an eternity all containers started working and i was able to access the nextcloud webui.

I am now able to see an appdata_* folder in the data dir, which was not present earlier, though there were all other files and folders like the admin folder or the .htaccess file. 
Not sure what caused this issue, but it is working now.

My guess is, one of the volumes or images were not deleted properly and therefore resulted in this issue, but i cannot tell for sure.

**Reply by szaimen** [reply]
> My guess is, one of the volumes or images were not deleted properly and therefore resulted in this issue

This is also my guess...
